tests/test_generate_rule_catalog.py

```
@@ -0,0 +1,362 @@
+            capsys: Pytest capture fixture for stdout/stderr.
+        """
+        import builtins
+
+        md = tmp_path / "rule.md"
+        md.write_text("---\nrule_id: L001\ndescription: Test rule\n---\n")
+
+        real_import = builtins.__import__
+
+        def fake_import(
+            name: str,
+            globals: Mapping[str, object] | None = None,
```

Ruff (0.15.21)

[error] Function argument `globals` is shadowing a Python builtin

(A002)